### ISSUE TITLE
Use Support Annotations

### ISSUE THREAD

**Issue by anticafe** [mikepenz/MaterialDrawer#491]
I see [Android Support Annotations](http://tools.android.com/tech-docs/support-annotations) is a really useful tools to help us avoid errors when coding.

In your code, I think there are many places we can utilize support annotations, for example, in `DrawerBuilder.java`:
- Method `public DrawerBuilder(Activity activity)`: can annotate with `(@NonNull Activity activity)`
- Method `public DrawerBuilder withRootView(int rootViewRes)`: can annotate with `(@LayoutRes int rootViewRes)`
- Method `public DrawerBuilder withStatusBarColorRes(int statusBarColorRes)`: can annotation `(@ColorRes int statusBarColorRes)`
- Method `public DrawerBuilder withStatusBarColor(int statusBarColor)`: can annotate with `(@ColorInt int statusBarColor)`

To sum up, there are many annotations can be utilized, such as: `@Nullable`, `@NonNull`, `@StringRes`, `@IntRes`, `@DrawableRes`, `@ColorRes`, `@LayoutRes`.

__EDIT__: I forget to mention that we can also integrate annotation into getter method results. For example, in `Drawer.java`:
- Method `public View getHeader()`: can annotate `@Nullable public View getHeader()`

Let us know if you can integrate them into your library :)

**Comment by mikepenz** [comment]
@anticafe will be in v4.0.0
